pdr_backend/subgraph/subgraph_predictions.py

```diff
@@ -112,12 +113,19 @@ def fetch_filtered_predictions(
                 }}
             }}"""
 
-        print("Querying subgraph...", query)
-        result = query_subgraph(
-            get_subgraph_url(network),
-            query,
-            timeout=20.0,
-        )
+        try:
+            print("Querying subgraph...", query)
+            result = query_subgraph(
+                get_subgraph_url(network),
+                query,
+                timeout=20.0,
+            )
+        except Exception as e:
+            print(
```

Comment:
```
Currently the function returns the records as if it was successful.
```
Currently the function prints the error, and returns all records that were previously processed succesfully.

```
Shouldn't we try to fetch the rest of the records in case of an error? 
```
There are a variety of errors that happen when fetching from subgraph... from the subgraph being down, to just a run-time error that enables you to resume right-away. Due to the range of errors that can happen, my initial thought was to just end the fetch/sync step, write everything that was successful to disk, and then the user can figure out next steps.

So:
1. First implementation (previously): error stops pipeline. Nothing is saved.
2. Second implementation (now): error stops pipeline. Everything that was successfully fetched, is saved out. User resumes accordingly.
3. Third implementation (next): error is handled correctly. If possible, system keeps going until everything has completed.

We could skip (2) and go straight towards handling all the errors (3), this just seemed a bit optimistic as we're still identifying issues with subgraph/sapphire.

Reply:
```
On error; automatically save the data fetched up to that point, mark the location of the error, and then later attempt to resume the fetching process from that marked point onward.
```

@trizin that's exactly what's happening with this implementation